cpp/src/arrow/compare.cc

```diff
@@ -862,7 +862,11 @@ class TypeEqualsVisitor {
 
 class ScalarEqualsVisitor {
  public:
-  explicit ScalarEqualsVisitor(const Scalar& right) : right_(right), result_(false) {}
+  explicit ScalarEqualsVisitor(const Scalar& right)
+      : right_(right), result_(false), options_(EqualOptions()) {}
+
+  explicit ScalarEqualsVisitor(const Scalar& right, const EqualOptions& opts)
```

Comment:
`const EqualOptions& opt = EqualOptions::Defaults()`

Reply:
Good suggestion. Thank you.